Reorg main readme make it more engaging. (#182)
opendatahub-io/llama-stack-demos: README.md

# Llama Stack Demos on OpenDataHub

This repository contains practical examples and demos designed to get you started quickly building AI apps with [Llama Stack](https://github.com/meta-llama/llama-stack) on Kubernetes or OpenShift. Whether you're a cluster admin looking to deploy the right GenAI infrastructure or a developer eager to innovate with AI Agents, the content in this repo should help you get started.

## 🛠️ Get Started

### Requirements

> To run these demos in their full glory, ensure your environment meets the following:

Reviewer: ```suggestion
To run these demos, ensure your environment meets the following:
```

* OpenShift Cluster 4.17+
* 2 GPUs with a minimum of 40GB VRAM each.

### Deployment Instructions

> Ready to launch? Follow these simple steps to deploy the core components:

Reviewer: ```suggestion
Next, follow these simple steps to deploy the core components:
```

> # 👋 Welcome to the Llama Stack Demos Repository! Unleash the Power of Advance AI agent on Kubernetes & OpenShift

Reviewer: ```suggestion
#  Llama Stack Demos on OpenDataHub
```

> Ready to build cutting-edge AI applications with [Llama Stack](https://github.com/meta-llama/llama-stack) on Kubernetes or OpenShift? You've come to the right place! This repository is your launchpad, packed with practical examples and demos designed to get you started quickly, whether you're a cluster admin looking to deploy or an AI developer eager to innovate.

Reviewer: ```suggestion
This repository contains practical examples and demos designed to get you started quickly building AI apps with [Llama Stack](https://github.com/meta-llama/llama-stack) on Kubernetes or OpenShift. Whether you're a cluster admin looking to deploy the right GenAI infrastructure or a developer eager to innovate with AI Agents, the content in this repo should help you get started. 
```

## ✨ What You'll Discover Here

Explore the exciting capabilities of Llama Stack through our key demos:

> * **Intelligent Agents Unleashed (RAG + Agent + MCP):** Dive into building sophisticated AI agents that understand documents, reason, invoke tools, and perform tasks using [Retrieval-Augmented Generation (RAG)](https://www.ibm.com/think/topics/retrieval-augmented-generation), [Model Context Protocol (MCP)](https://github.com/modelcontextprotocol), and advanced agentic workflows.
> * **Effortless Deployment & Monitoring:** Learn how to seamlessly deploy these powerful agents on Kubernetes & OpenShift using the provided manifests and monitor their performance with our integrated observability toolkit.

Reviewer: Can we take this part out? I'm not sure what its trying to convey

```suggestion
```

## 💡 Demo Architecture
Imagine a secure Llama Stack application running on OpenShift, powered by MCP tools and a [Milvus](https://milvus.io/) vector database for intelligent RAG and agentic workflows. This architecture showcases the seamless integration of these powerful components.

![Architecture Diagram](./images/architecture-diagram.jpg)

> 

Reviewer: Can we revert to the old text here? Also, lets link to the demo/rag_agentic this is the diagram for.

> ## 🛠️ Get Started: Dive into Llama Stack Demos

Reviewer: ```suggestion
## 🛠️ Get Started
```

> This repository offers tailored resources to kickstart your journey with Llama Stack:

Reviewer: ```suggestion

```

### For AI Innovators (Builders of Intelligent Applications)

1.  **Your Starting Point: The RAG/Agentic Demo Guide:** Begin your journey with the comprehensive [RAG/Agentic README](./demos/rag_agentic/README.md). This guide will introduce you to the core concepts and then naturally lead you to the progressive **notebooks** for hands-on learning, frontend documentation and some source code.
2.  **Explore Collaborative AI (Beta):** Delve deeper into agent-to-agent communication by exploring the [A2A implementation](./demos/a2a_llama_stack/) after you've grasped the basics in the RAG/Agentic demo.
3.  **Understand Evaluation:** Check out the [tests/](./tests/) directory in the main repository to learn about our evaluation methodologies for LLM and MCP tool combinations, giving you insights into building high-quality AI solutions.

> 

Reviewer: Lets delete all this content. Move the "get started" header to just after the first paragraph. and put the content in 
"Ready to Deploy? Here's What You'll Need " there at the top.

## ⚙️ Ready to Deploy? Here's What You'll Need

To run these demos in their full glory, ensure your environment meets the following:

* OpenShift Cluster 4.17+
* 2 GPUs with a minimum of 40GB VRAM each.

> ### 🚀 Deployment Instructions

Reviewer: This whole section should be moved to the top based on my above comment.

Ready to launch? Follow these simple steps to deploy the core components, including those required for the "ultimate agent" demo:

1.  Create a dedicated OpenShift project:
    ```bash
    oc new-project llama-serve
    ```
2.  Apply the Kubernetes manifests:
    ```bash
    oc apply -k kubernetes/kustomize/overlay/all-models
    ```
    This will deploy the foundational Llama Stack services, vLLM model servers, and MCP tool servers, enabling the "ultimate agent" demo.

### 🐍 Setting Up Your Development Environment

We use `uv` for managing Python dependencies, ensuring a consistent and efficient development experience. Here's how to get your environment ready:

1.  Install `uv`:
    ```bash
    pip install uv
    ```
2.  Synchronize your environment:
    ```bash
    uv sync
    ```
3.  Activate the virtual environment:
    ```bash
    source .venv/bin/activate
    ```

Now you're all set to run any Python scripts or Jupyter notebooks within the `demos/rag_agentic` directory!

---

> We're excited to see what you build with Llama Stack! If you have any questions or feedback, please don't hesitate to open an issue. Happy building! 🎉

Reviewer: ```suggestion
We're excited to see what you build with Llama Stack! If you have any questions or feedback, please don't hesitate to open an [issue](https://github.com/opendatahub-io/llama-stack-demos/issues). Happy building! 🎉
```
